ts/src/bingx.ts

```diff
@@ -0,0 +1,2342 @@
+         * @see https://bingx-api.github.io/docs/#/swapV2/market-api.html#Get%20Market%20Depth
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int|undefined} limit the maximum amount of order book entries to return
+         * @param {object} params extra parameters specific to the bingx api endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/#/?id=order-book-structure} indexed by market symbols
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        let response = undefined;
+        let marketType = undefined;
+        [ marketType, params ] = this.handleMarketTypeAndParams ('fetchOrderBook', market, params);
+        if (marketType === 'spot') {
+            response = await this.spotV1PublicGetMarketDepth (this.extend (request, params));
+        } else {
+            response = await this.swapV2PublicGetQuoteDepth (this.extend (request, params));
+        }
+        //
+        // spot
+        //
+        //     {
+        //         "code": 0,
+        //         "data": {
+        //           "bids": [
+        //             [
+        //               "26324.73",
+        //               "0.37655"
+        //             ],
+        //             [
+        //               "26324.71",
+        //               "0.31888"
+        //             ],
+        //         ],
+        //         "asks": [
+        //             [
+        //               "26340.30",
+        //               "6.45221"
+        //             ],
+        //             [
+        //               "26340.15",
+        //               "6.73261"
+        //             ],
+        //         ]}
+        //     }
+        //
+        // swap
+        //
+        //     {
+        //         "code": 0,
+        //         "msg": "",
+        //         "data": {
+        //           "T": 1683914263304,
+        //           "bids": [
+        //             [
+        //               "26300.90000000",
+        //               "30408.00000000"
+        //             ],
+        //             [
+        //               "26300.80000000",
+        //               "50906.00000000"
+        //             ],
+        //         ],
+        //         "asks": [
+        //             [
+        //               "26301.00000000",
+        //               "43616.00000000"
+        //             ],
+        //             [
+        //               "26301.10000000",
+        //               "49402.00000000"
+        //             ],
+        //         ]}
+        //     }
+        //
+        const orderbook = this.safeValue (response, 'data', {});
+        const timestamp = this.safeInteger (orderbook, 'T');
```

Comment:
we have the spot timestamp available in the `ts` field 
![image](https://github.com/ccxt/ccxt/assets/43336371/86bc02df-6652-4147-8ded-9f63bb27a755)